src/parameters_parser.cc

```diff
@@ -7,11 +7,26 @@ namespace Kakoune
 
 String generate_switches_doc(const SwitchMap& switches)
 {
+    Vector<int> lengths(switches.size());
```

Comment:
I would avoid storing lengths, they are simple to compute, I suggest to write a small named lambda and use ranges:
```
auto switch_len = [](auto& sw) { return sw.key.length() + (sw.value.take_arg ? 5 : 0); };
auto switches_len = switches | transform(switch_len);
const CharCount maxlen = *std::max_element(switches_len.begin(), switches_len.end());
```

Reuse that switch_len lambda in the second loop to get the len. I believe the extra computations will be largely cheaper than allocating a vector.